Source/JavaScriptCore/runtime/JSONObject.cpp

```diff
@@ -627,6 +637,517 @@ bool Stringifier::Holder::appendNextProperty(Stringifier& stringifier, StringBui
+        if (remainingCapacity() < 5) {
+            recordBufferFull();
+            return;
+        }
+        m_buffer[m_length] = 'f';
+        m_buffer[m_length + 1] = 'a';
+        m_buffer[m_length + 2] = 'l';
+        m_buffer[m_length + 3] = 's';
+        m_buffer[m_length + 4] = 'e';
+        m_length += 5;
+        return;
+    }
+
+    if (value.isInt32()) {
+        auto number = value.asInt32();
+        auto length = lengthOfIntegerAsString(number);
+        if (remainingCapacity() < length) {
+            recordBufferFull();
+            return;
+        }
+        writeIntegerToBuffer(number, &m_buffer[m_length]);
+        m_length += length;
+        return;
+    }
+
+    if (!value.isCell()) {
+        recordFailure("value type"_s);
+        return;
+    }
+    auto& cell = *value.asCell();
+
+    switch (cell.type()) {
+    case StringType: {
+        auto& string = asString(&cell)->tryGetValue();
+        if (string.isNull()) {
+            recordFailure("String::tryGetValue"_s);
+            return;
+        }
+        if (!string.is8Bit()) {
+            recordFailure("16-bit string"_s);
+            return;
+        }
+        auto stringLength = string.length();
+        if (remainingCapacity() < 1 + stringLength + 1) {
+            recordBufferFull();
+            return;
+        }
+        m_buffer[m_length] = '"';
+        auto* characters = string.characters8();
+        for (unsigned i = 0; i < stringLength; ++i) {
+            auto character = characters[i];
+            if (WTF::escapedFormsForJSON[character]) {
+                recordFailure("string character needs escaping"_s);
+                return;
+            }
+            m_buffer[m_length + 1 + i] = character;
+        }
+        m_buffer[m_length + 1 + stringLength] = '"';
+        m_length += 1 + stringLength + 1;
+        return;
+    }
+
+    case ObjectType:
+    case FinalObjectType: {
+        auto& object = *asObject(&cell);
+        if (object.isCallable()) {
+            recordFailure("callable object"_s);
+            return;
+        }
+        auto* ancestorClass = object.classInfo();
+        do {
+            if (ancestorClass == NumberObject::info()) {
+                recordFailure("wrapped number"_s);
+                return;
+            }
+            if (ancestorClass == StringObject::info()) {
+                recordFailure("wrapped string"_s);
+                return;
+            }
+            if (ancestorClass == BooleanObject::info()) {
+                recordFailure("wrapped boolean"_s);
+                return;
+            }
+            if (ancestorClass == BigIntObject::info()) {
+                recordFailure("wrapped big int"_s);
+                return;
+            }
+        } while ((ancestorClass = ancestorClass->parentClass));
+        auto& structure = *object.structure();
+        if (object.structureID() != structure.id()) {
+            recordFailure("structure ID mismatch"_s);
+            return;
+        }
```

Comment:
We do not need to have this check. This Structure check was existing only to capture structure transition from the previous operations.

Reply:
Thanks, removed locally.